stages/01_find_data.py

```diff
@@ -0,0 +1,27 @@
+# Imports
+# -------
+import urllib.request
+from bs4 import BeautifulSoup
+
+dude_target_page = urllib.request.urlopen("https://dude.docking.org/targets")
+dude_target_page = dude_target_page.read().decode('utf-8')
+
+soup = BeautifulSoup(dude_target_page, 'lxml')
+target_table = soup.find('table')
+rows = target_table.findAll('tr')
+
+active_base_url = 'https://dude.docking.org//targets/PROTEIN/actives_nM_combined.ism\n'
+inactive_base_url = 'https://dude.docking.org//targets/PROTEIN/inactives_nM_combined.ism\n'
+
+links_text = open('links.txt', 'w')
+
+for row in rows:
+    cols = row.find_all('td')
+    if cols:
+        protein_target = cols[1].get_text().lower()
+
+        active_url = active_base_url.replace('PROTEIN', protein_target)
+        inactive_url = inactive_base_url.replace('PROTEIN', protein_target)
+
+        links_text.write('wget -P download/' + protein_target + '_active.ism' + ' ' + active_url)
+        links_text.write('wget -P download/' + protein_target + '_inactive.ism' + ' ' + inactive_url)
```

Comment:
Could you give me information on which specific files you want to download? Are the tarballs for each individual target not sufficient? There may be a way to use `wget` alone to get everything.

Reply:
Yeah. 

So for the reasoning and we can chat later today. The original DUDE docking dataset is 2.6 GB and it's got an `all.tar.gz` file in there. When that happens and you download the file the chemicals and coordinates are present however not the inhibition numbers per molecule. 

So I went to the target page on DUDE: 

<img width="1201" alt="Screenshot 2024-07-19 at 10 26 59 AM" src="https://github.com/user-attachments/assets/aaeedac2-ba94-4274-872d-e681b6d5ce1f">

When you click a link:

<img width="1162" alt="Screenshot 2024-07-19 at 10 27 31 AM" src="https://github.com/user-attachments/assets/2cac7fd0-6c14-4027-ab3e-b9c6996c6a24">

I think adding the DUDE should be done in stages. First the most valuable data is the ligands -> receptor -> binding affinity values for the `active` compounds and the `inactive` compounds. This is most useful for ML models downstream. 

We also should extract the `receptor.pdb` file since we need the protein correlation. Coordinates can be generated algorithmithically and not useful data because it most likely won't be resused. 

Other data that can be handled later is these two:

<img width="434" alt="Screenshot 2024-07-19 at 10 30 09 AM" src="https://github.com/user-attachments/assets/080a0904-88ee-409b-819a-f5720e505a50">

<img width="479" alt="Screenshot 2024-07-19 at 10 30 11 AM" src="https://github.com/user-attachments/assets/2bf4d501-0397-4ce8-a481-bfa510fab014">

Which is not most useful data and most likely would not be used. Enumeration is based on their own internal algorithm and can also be generated with RDKit.

Reply:
Yep, so as we discussed, the `download` dvc stage can go ahead and download as much of the current data as possible, and for the first go at processing that data into Parquet, select the files that you think are most important for any downstream use. If a number of the `*.ism` files all contain the same layout of data, then they can be processed at once.

I also recommend taking a look at the input and output data inside of [VisiData](https://www.visidata.org/install/) as you can use that to browse the data to make sure it got created correctly.